src/Microsoft.OData.Core/JsonLight/ODataJsonLightWriter.cs

```diff
@@ -1018,32 +1021,51 @@ protected override void WriteEntityReferenceInNavigationLinkContent(ODataNestedR
             // In JSON Light, we can only write entity reference links at the beginning of a navigation link in requests;
             // once we wrote a resource set, entity reference links are not allowed anymore (we require all the entity reference
             // link to come first because of the grouping in the JSON Light wire format).
-            JsonLightNestedResourceInfoScope navigationLinkScope = (JsonLightNestedResourceInfoScope)this.CurrentScope;
-            if (navigationLinkScope.ResourceSetWritten)
+            JsonLightNestedResourceInfoScope nestedResourceScope = this.CurrentScope as JsonLightNestedResourceInfoScope;
+            if (nestedResourceScope == null)
+            {
+                nestedResourceScope = this.ParentNestedResourceInfoScope as JsonLightNestedResourceInfoScope;
+            }
+
+            if (nestedResourceScope.ResourceSetWritten)
             {
                 throw new ODataException(Strings.ODataJsonLightWriter_EntityReferenceLinkAfterResourceSetInRequest);
             }
 
-            if (!navigationLinkScope.EntityReferenceLinkWritten)
+            if (!nestedResourceScope.EntityReferenceLinkWritten)
             {
-                if (!this.jsonLightOutputContext.WritingResponse &&
-                    (this.Version == null || this.Version < ODataVersion.V401))
-                {
-                    // Write the property annotation for the entity reference link(s)
-                    this.odataAnnotationWriter.WritePropertyAnnotationName(parentNestedResourceInfo.Name, ODataAnnotationNames.ODataBind);
-                }
-                else
+                // In request

```

Comment:
>In request [](start = 19, length = 10)

I understand writing a different annotation in a request for 4.0 versus 4.01 and greater, but not sure why the logic for writing the open/close square bracket varies from request to response.

Reply:
in Request:
```
WriteStart( nestedResourceInfo ) // collection
      WriteEntityReferenceLink( EntityReferenceLink)
      WriteEntityReferenceLink( EntityReferenceLink)
      WriteEntityReferenceLink( EntityReferenceLink)
WriteEnd() // end of nested resource info
```
the open square bracket is written in first call to : "WriteEntityReferenceLink(..)

Meanwhile, in response, 

```
WriteStart( nestedResourceInfo ) // collection
   WriteStart( new ODataResourceSet() ) // Add a resource set writing logic
      WriteEntityReferenceLink( EntityReferenceLink)
      WriteEntityReferenceLink( EntityReferenceLink)
      WriteEntityReferenceLink( EntityReferenceLink)
  WriteEnd() // end of resource set
WriteEnd() // end of nested resource info
```

the open square bracket is written in the "WriteStart(new ODataResourceSet()), so, when we call the first "WriteEntityReferenceLink(...)", it needn't to write the open square bracket. 

Same for the close square bracket.



---
In reply to: [285803927](https://github.com/OData/odata.net/pull/1464#discussion_r285803927) [](ancestors = 285803927)

Reply:
Yech. As discussed, we need to fix this to support the same pattern in request and response (the new one) and deprecate the old pattern.

---
In reply to: [289081395](https://github.com/OData/odata.net/pull/1464#discussion_r289081395) [](ancestors = 289081395,285803927)